implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OTelCollectorIntegrationTest.java

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.registry.otlp;
+
+import io.micrometer.core.instrument.*;
+import io.restassured.response.Response;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.time.Duration;
+
+import static io.restassured.RestAssured.given;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.*;
+import static org.testcontainers.containers.BindMode.READ_ONLY;
+import static uk.org.webcompere.systemstubs.SystemStubs.withEnvironmentVariables;
+
+/**
+ * Integration tests for {@link OtlpMeterRegistry} and the OTel collector
+ *
+ * @author Jonatan Ivanov
+ */
+@Testcontainers
+@Tag("docker")
+class OTelCollectorIntegrationTest {
+
+    private static final String OPENMETRICS_100 = "application/openmetrics-text; version=1.0.0; charset=utf-8";
+
+    private static final String CONFIG_FILE_NAME = "collector-config.yml";
+
+    @Container
+    private final GenericContainer<?> container = new GenericContainer("otel/opentelemetry-collector-contrib")
+        .withCommand("--config=/etc/" + CONFIG_FILE_NAME)
+        .withClasspathResourceMapping(CONFIG_FILE_NAME, "/etc/" + CONFIG_FILE_NAME, READ_ONLY)
+        .withExposedPorts(4318, 9090); // HTTP receiver, Prometheus exporter
+
+    @Test
+    void collectorShouldExportMetrics() throws Exception {
+        MeterRegistry registry = createOtlpMeterRegistryForContainer(container);
+        Counter.builder("test.counter").register(registry).increment(42);
+        Gauge.builder("test.gauge", () -> 12).register(registry);
+        Timer.builder("test.timer").register(registry).record(Duration.ofMillis(123));
+        DistributionSummary.builder("test.distributionsummary").register(registry).record(24);
+
+        await().atMost(Duration.ofSeconds(5))
+            .pollDelay(Duration.ofMillis(100))
+            .pollInterval(Duration.ofMillis(100))
+            .untilAsserted(() -> whenPrometheusScraped().then().body(not(empty())));
+
+        // @formatter:off
+        // tags can vary depending on where you run your tests:
+        //  - IDE: no telemetry_sdk_version tag
+        //  - Gradle: telemetry_sdk_version has the version number
+        whenPrometheusScraped().then().body(
+            containsString("{job=\"test\",service_name=\"test\",telemetry_sdk_language=\"java\",telemetry_sdk_name=\"io.micrometer\""),
+
+            matchesPattern("(?s)^.*test_counter\\{.+} 42\\n.*$"),
+            matchesPattern("(?s)^.*test_gauge\\{.+} 12\\n.*$"),
+
+            matchesPattern("(?s)^.*test_timer_count\\{.+} 1\\n.*$"),
+            matchesPattern("(?s)^.*test_timer_sum\\{.+} 123\\n.*$"),
```

Comment:
Now that I'm looking at this, I don't think this is correct, this should be 123ms not 123s.

Reply:
Seems like a collector issue, the data that we send looks like this:
```json
metrics {
  name: "test.timer"
  unit: "milliseconds"
  histogram {
    data_points {
      start_time_unix_nano: 1677210838494000000
      time_unix_nano: 1677210839021000000
      count: 1
      sum: 123.0
    }
    aggregation_temporality: AGGREGATION_TEMPORALITY_CUMULATIVE
  }
}
```

Reply:
Created an issue to track this: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18903

Reply:
Good catch. I guess we'll need a fix in the collector before we can change the test.

Reply:
It seems like it might be intentional that no conversion is done currently. But then there's no way to know the unit as far as I can tell.
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/translator/prometheus

Reply:
I'm wondering if there are receivers that mandate a unit that is not seconds.
Or two receivers that mandate different units. Or two exporters with different units.
In all of these cases I think not doing the conversion can lead to wrong result.